plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1183,18 +1191,46 @@ def get_transaction_receipt(
         "Error when attempting getting tx receipt: {}", logger_method="debug"
     )
     def _try_get_transaction_receipt(
-        self, tx_digest: str, **_kwargs: Any
+        self,
+        tx_digest: str,
+        timeout: Optional[int] = None,
+        poll_latency: Optional[float] = None,
+        **_kwargs: Any,
     ) -> Optional[JSONLike]:
         """
         Try get the transaction receipt.
 
         :param tx_digest: the digest associated to the transaction.
+        :param timeout: time to wait before assuming the receipt is not available, defaults to infinite.
+        :param poll_latency: backoff between retries, if None, a single try (call) will be made.
         :param _kwargs: the keyword arguments. Possible kwargs are:
             `raise_on_try`: bool flag specifying whether the method will raise or log on error (used by `try_decorator`)
         :return: the tx receipt, if present
         """
-        tx_receipt = self._api.eth.get_transaction_receipt(  # pylint: disable=no-member
-            cast(HexStr, tx_digest)
+        if timeout is None and poll_latency is None:
+            # if timeout and poll_latency have not been provided
+            # we make a single request
+            tx_receipt = (
+                self._api.eth.get_transaction_receipt(  # pylint: disable=no-member
+                    cast(HexStr, tx_digest),
+                )
+            )
+            return AttributeDictTranslator.to_dict(tx_receipt)
+
+        # either a polling interval or a timeout was provided
+        if timeout is None:
+            # if not provided, we wait for ~31 years.
+            timeout = 10**9
```

Comment:
what? why would we do that?

Reply:
If a timeout is not provided we assume that the user wants to wait indefinitely. Although 31 years is not infinity, it should be more than enough :laughing:

Reply:
Looks hacky, but I don't see how we could keep backward compatibility otherwise.
Let's at least create a constant and add a comment if we keep this.